src/common/hooks/useUploadFileInChunks.js

```diff
@@ -0,0 +1,117 @@
+import { useState, useCallback } from 'react';
+import bc from '../services/breathecode';
+import { getStorageItem } from '../../utils';
+
+const useUploadFileInChunks = () => {
+  const [isSplitting, setIsSplitting] = useState(false);
+  const [isUploadingChunks, setIsUploadingChunks] = useState(false);
+  const [isFinalizing, setIsFinalizing] = useState(false);
+  const [error, setError] = useState(null);
+
+  const getOperationMeta = async (operationType) => {
+    const response = await bc.media().operationMeta(operationType);
+    return response.data;
+  };
+
+  const splitFileIntoChunks = (file, chunkSize) => {
+    const chunks = [];
+    let start = 0;
+
+    while (start < file.size) {
+      const end = Math.min(start + chunkSize, file.size);
+      const chunk = file.slice(start, end, file.type);
+      chunks.push(chunk);
+      start = end;
+    }
+    return chunks;
+  };
+
+  const uploadChunk = async (file, chunk, operationType, totalChunks, chunkIndex, academyID = undefined) => {
+    try {
+      const prefix = academyID ? 'academy/chunk' : 'me/chunk';
+      const formData = new FormData();
+      formData.append('operation_type', operationType);
+      formData.append('filename', file.name);
+      formData.append('total_chunks', totalChunks);
+      formData.append('chunk', chunk);
+      formData.append('chunk_index', chunkIndex);
+
+      const accessToken = getStorageItem('accessToken');
+      if (!accessToken) {
+        console.error('No se encontró el accessToken. Asegúrate de que esté almacenado correctamente.');
+        return null;
+      }
+
+      const headers = academyID ? { Academy: academyID } : { Authorization: `Token ${accessToken}` };
+      const response = await bc.media().uploadChunk(prefix, formData, headers);
+
+      return response;
+    } catch (err) {
+      console.error(`Error uploading chunk ${chunkIndex}:`, err);
+      throw err;
+    }
+  };
+
+  const endFileUpload = async (operationType, totalChunks, filename, mime, meta, academyID = undefined) => {
+    const prefix = academyID ? 'academy/chunk/upload' : 'me/chunk/upload';
+    const response = await bc.media().endFileUpload(prefix, {
+      operation_type: operationType,
+      total_chunks: totalChunks,
+      filename,
+      mime,
+      meta,
+    });
+
+    return response.data;
+  };
+
+  const uploadFileInChunks = useCallback(async (file, operationType, meta, academyID = undefined) => {
```

Comment:
Why are you using useCallback here? Is it a page where we are rendering and using the same info?